packages/stack-shared/src/known-errors.tsx

```diff
@@ -1218,6 +1218,48 @@ const InvalidPollingCodeError = createKnownErrorConstructor(
   (json: any) => [json] as const,
 );
 
+
+const InvalidApiKey = createKnownErrorConstructor(
+  KnownError,
+  "INVALID_API_KEY",
+  () => [
+    401,
+    "The provided API key is invalid.",
+  ] as const,
+  () => [] as const,
+);
+
+const ApiKeyExpired = createKnownErrorConstructor(
+  KnownError,
+  "API_KEY_EXPIRED",
+  () => [
+    401,
+    "API key has expired.",
+  ] as const,
+  () => [] as const,
+);
+
+const ApiKeyRevoked = createKnownErrorConstructor(
+  KnownError,
+  "API_KEY_REVOKED",
+  () => [
+    401,
+    "API key has been revoked.",
+  ] as const,
+  () => [] as const,
+);
+
+const WrongApiKeyType = createKnownErrorConstructor(
+  KnownError,
+  "WRONG_API_KEY_TYPE",
+  (expectedType: string, actualType: string) => [
+    400,
+    `This endpoint is for ${expectedType} API keys, but a ${actualType} API key was provided.`,
+    { expected_type: expectedType, actual_type: actualType },
+  ] as const,
+  (json) => [json.expected_type, json.actual_type] as const,
+);
```

Comment:
All of these could inherit another type `InvalidApiKey`, and then in when you use them you can just check for `instanceof ApiKeyNotValid`

```suggestion
const ApiKeyNotValid = createKnownErrorConstructor(
  KnownError,
  "API_KEY_NOT_VALID",
  "inherit",
  "inherit",
);

const ApiKeyExpired = createKnownErrorConstructor(
  ApiKeyNotValid,
  "API_KEY_EXPIRED",
  () => [
    401,
    "API key has expired.",
  ] as const,
  () => [] as const,
);

const ApiKeyRevoked = createKnownErrorConstructor(
  ApiKeyNotValid,
  "API_KEY_REVOKED",
  () => [
    401,
    "API key has been revoked.",
  ] as const,
  () => [] as const,
);

const WrongApiKeyType = createKnownErrorConstructor(
  ApiKeyNotValid,
  "WRONG_API_KEY_TYPE",
  (expectedType: string, actualType: string) => [
    400,
    `This endpoint is for ${expectedType} API keys, but a ${actualType} API key was provided.`,
    { expected_type: expectedType, actual_type: actualType },
  ] as const,
  (json) => [json.expected_type, json.actual_type] as const,
);
```